cmd/playground/window_model.go

```diff
@@ -0,0 +1,207 @@
+		// and start updating Sizeinput(hence Sizeinput.Focus())
+		m.offsetInput.Blur()
+		m.sizeInput.Focus()
+		m.cursor = pointToSize
+	case pointToTruncate:
+		m.offsetInput.Focus()
+		m.cursor = pointToOffset
+	case pointToYear, pointToMonth, PointToDay, pointToHour, pointToMinute:
+		m.cursor = pointToTruncate
+	}
+}
+
+func (m *model) handleDown() {
+	switch m.cursor {
+	case pointToTruncate:
+		m.cursor = pointToHour
+	case pointToOffset:
+		m.offsetInput.Blur()
+		m.cursor = pointToTruncate
+	case pointToSize:
+		m.offsetInput.Focus()
+		m.sizeInput.Blur()
+		m.cursor = pointToOffset
+	}
+}
+
+// handles left and right arrow key movements
+func (m *model) handleRight() {
+	switch m.cursor {
+	case pointToMonth:
+		m.cursor = pointToYear
+	case PointToDay:
+		m.cursor = pointToMonth
+	case pointToMinute:
+		m.cursor = PointToDay
+	case pointToHour:
+		m.cursor = pointToMinute
+	}
+}
+
+func (m *model) handleLeft() {
+	switch m.cursor {
+	case pointToMinute:
+		m.cursor = pointToHour
+	case PointToDay:
+		m.cursor = pointToMinute
+	case pointToMonth:
+		m.cursor = PointToDay
+	case pointToYear:
+		m.cursor = pointToMonth
+	}
+}
+func (m *model) handleIncrease() {
+	switch m.cursor {
+	case pointToTruncate:
+		m.state.IncrementTruncate()
+	case pointToYear, pointToMonth, PointToDay, pointToHour, pointToMinute:
+		m.state.IncrementTime(m.cursor.getStringValue())
+	}
+}
+func (m *model) handleDecrease() {
+	switch m.cursor {
+	case pointToTruncate:
+		m.state.DecrementTruncate()
+	case pointToYear, pointToMonth, PointToDay, pointToHour, pointToMinute:
+		m.state.DecrementTime(m.cursor.getStringValue())
+	}
+}
+
+func (m *model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	// switch currMsg := msg.(type) {
+	// case tea.KeyMsg:
+	currMsg := reflect.TypeOf(msg)
+	if currMsg.String() == "tea.KeyMsg" {
+		switch fmt.Sprintf("%s", msg) {
+		case "ctrl+c", "q":
+			return m, tea.Quit
+		case "up":
+			m.handleUp()
+		case "down":
+			m.handleDown()
+		case "left":
+			m.handleLeft()
+		case "right":
+			m.handleRight()
+		case "shift+up", "shift+right":
+			m.handleIncrease()
+		case "shift+down", "shift+left":
+			m.handleDecrease()
+		case "M", "h", "-", "1", "2", "3", "4", "5", "6", "7", "8", "9", "0", "backspace":
+			// to get updated the input fields must be focussed since we handle the Focus and Blur we can directly update them here
+			m.sizeInput, _ = m.sizeInput.Update(msg)
+			m.offsetInput, _ = m.offsetInput.Update(msg)
+			m.state.updateWindowparameters(m.sizeInput.Value(), m.offsetInput.Value())
+		}
+	}
+	return m, nil
+}
+
+func (m *model) genarateCursorIfPointed(position string, value string) string {
+	if m.cursor.getStringValue() == position {
+		var s strings.Builder
+		s.WriteString("[")
+		s.WriteString(value)
+		s.WriteString("]")
+		return s.String()
+	}
+	return " " + value
+}
+
+// we genare a string representing the sechduled time , which also adds a cursor if it is pointing to the any of the fields in sechdueld date
```

Comment:
avoid any subject like I, we, you, they, etc. Instead, use the name of the method like `genarateSechduledDateView generates a string ...`

Reply:
Also, the method name is typo, should be `generate...` not `genarate...`